### PR TITLE
Support Conv2dTranspose and GroupConv2dTranspose for IE

### PR DESCRIPTION
**Pull request by miaobin**
@fujunwei @mingmingtasd PTAL, thanks!
Test results:
[----------] Global test environment tear-down
[==========] 104 tests from 1 test suite ran. (340 ms total)
[  PASSED  ] 102 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] Conv2dTests.Conv2dTransposeWithAutoPadSameUpperDefault
[  FAILED  ] Conv2dTests.Conv2dTransposeWithAutoPadSameLowerNhwcIhwo

### CODE REVIEW

File: src/webnn_native/ops/Conv2d.cpp [hunk 1]
```diff
@@ -52,6 +52,22 @@ namespace webnn_native { namespace op {
...
+        mOptions.outputPadding = mOutputPadding.data();
+        mOptions.outputPaddingCount = mOutputPadding.size();
+
+        if (options != nullptr && options->outputSizes != nullptr) {
+            mOutputSizes.assign(options->outputSizes,
+                                options->outputSizes + options->outputSizesCount);
+            mOptions.outputSizes = mOutputSizes.data();
+            mOptions.outputSizesCount = mOutputSizes.size();
+        }
+        mOptions.transpose = options == nullptr ? false : options->transpose;
```
Comment (mingmingtasd) [inline src/webnn_native/ops/Conv2d.cpp:70]: Should we combine L70-L79 into scope of `if (!options){}`

**Comment by miaobin** [comment]
Reported a [issue](https://github.com/openvinotoolkit/openvino/issues/7321) to openvino upstream.

File: src/tests/end2end/Conv2dTests.cpp [hunk 2]
```diff
@@ -1252,4 +1252,657 @@ TEST_F(Conv2dTests, Conv2dWithAutoPadSameUpperNhwcIhwo) {
...
+    Tensor filter = {{3, 3, 1, 2}, std::vector<float>(18, 1)};
+    Tensor expected = {
+        {1, 2, 5, 5},
+        {0.,  1.,  3.,  3.,  2.,  3.,  8.,  15., 12., 7.,  9.,  21., 36., 27., 15., 9.,  20.,
+         33., 24., 13., 6.,  13., 21., 15., 8.,  0.,  1.,  3.,  3.,  2.,  3.,  8.,  15., 12.,
+         7.,  9.,  21., 36., 27., 15., 9.,  20., 33., 24., 13., 6.,  13., 21., 15., 8.},
+    };
+    utils::Conv2dOptions options;
+    options.transpose = true;
+    options.inputLayout = ml::InputOperandLayout::Nchw;
```
[the remark this answers is not in the document]
Comment (miaobin) [reply src/tests/end2end/Conv2dTests.cpp:1282]: This is a test for the input and output dimensions. I think it is necessary for us to write it here to show the intention explicitly. What do you think about it?

[the remark this answers is not in the document]
Comment (mingmingtasd) [reply src/tests/end2end/Conv2dTests.cpp:1282]: OK, thanks.

File: src/webnn_native/ops/Conv2d.cpp (hunk 1 above)
Comment (miaobin) [reply src/webnn_native/ops/Conv2d.cpp]: Already optimized. PTAL, thanks!

**Comment by miaobin** [comment]
Rebase the feature branch onto master.

File: src/webnn_native/openvino/GraphIE.cpp [hunk 3]
```diff
@@ -524,13 +546,26 @@ namespace webnn_native { namespace ie {
         DAWN_ASSERT(padding.size() == 4);
         std::vector<size_t> dilations(options->dilations,
                                       options->dilations + options->dilationsCount);
-        DAWN_ASSERT(strides.size() == 2);
+        DAWN_ASSERT(dilations.size() == 2);
+        std::vector<int32_t> outputPadding(options->outputPadding,
+                                           options->outputPadding + options->outputPaddingCount);
+        DAWN_ASSERT(outputPadding.size() == 2);
+        ngraph_node_t* outputShapeNode = nullptr;
+        if (options->outputSizes != nullptr) {
```
Comment (fujunwei) [inline src/webnn_native/openvino/GraphIE.cpp:554]: It's better to be used in transpose case like
```
if (transpose && options->outputSizes != nullptr) {
}
```

File: src/webnn_native/openvino/GraphIE.cpp [hunk 4]
```diff
@@ -173,27 +173,49 @@ namespace webnn_native { namespace ie {
...
+                    case ml::FilterOperandLayout::Ihwo:
+                        order = std::vector<int64_t>{0, 3, 1, 2};
+                        break;
+                    default:
+                        WEBNN_ASSERT(0, "The filter layout isn't supported.");
+                        break;
+                }
+            } else {
+                // FilterOperandLayout => oihw
+                if (layout == ml::FilterOperandLayout::Oihw) {
```
Comment (fujunwei) [inline src/webnn_native/openvino/GraphIE.cpp:202]: Move it to L.205
```
case ml::FilterOperandLayout::Oihw:
    return const_cast<ngraph_node_t*>(node);
```

File: src/webnn_native/openvino/GraphIE.cpp (hunk 3 above)
Comment (miaobin) [reply src/webnn_native/openvino/GraphIE.cpp]: Optimized.

File: src/webnn_native/openvino/GraphIE.cpp (hunk 4 above)
Comment (miaobin) [reply src/webnn_native/openvino/GraphIE.cpp]: Optimized and rebased the branch onto main.

**Review by fujunwei — APPROVED** [review]
LGTM.

**Comment by fujunwei** [comment]
Let's merge it, thanks.

**Pull request merged**